### PR TITLE
Add a refreshing auth token for in-cluster configurations.

### LINKED ISSUE #317: client should periodically reload InClusterConfig token
See https://github.com/kubernetes/kubernetes/issues/70679

### PR DESCRIPTION
**Pull request by brendandburns**
Closes #317

### CODE REVIEW

File: src/file_auth.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+export class FileAuth implements Authenticator {
+    private token: string | null = null;
+    private lastRead: Date | null = null;
+
+    public isAuthProvider(user: User): boolean {
+        return user.authProvider && user.authProvider.config && user.authProvider.config.tokenFile;
+    }
+
+    public getToken(user: User): string | null {
```
Comment (silasbw) [inline src/file_auth.ts]: I noticed https://github.com/kubernetes-client/javascript/blob/master/src/auth.ts#L8.  Is it worth rolling this `getToken` implementation into `applyAuthentication` now?

File: src/file_auth.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            this.readToken(user.authProvider.config.tokenFile);
+        }
+        return this.token;
+    }
+
+    public async applyAuthentication(user: User, opts: request.Options | https.RequestOptions) {
+        // nothing to do.
+    }
+
+    private readToken(filePath: string) {
```
Comment (silasbw) [inline src/file_auth.ts]: renaming nit: `readToken` to `updateToken`

I'd expect `readToken` to return something read.  `updateToken` implies updating some state.

Comment (drubin) [reply src/file_auth.ts]: `refreshToken` feels like a better name for this.

File: src/config.ts [hunk 3]
```diff
@@ -190,7 +192,12 @@ export class KubeConfig {
         this.users = [
             {
                 name: userName,
-                token: fs.readFileSync(`${pathPrefix}${Config.SERVICEACCOUNT_TOKEN_PATH}`).toString(),
```
Comment (drubin) [inline src/config.ts]: @brendanburns What happens in the case that still is still a valid form of authentication?

Does it feel like we are removing this functionality of being able to raw pass just a token we should be updating more tests/code to clean up the usage of `user.token`

**Pull request merged**